processing/format_qaqc_driver.py

```diff
@@ -0,0 +1,338 @@
+                self.timeout = self.max_timeout / 10.0
+                self.lookback_h = config.getint(cfg_section, 'lookback_h')
+
+            cfg_section = 'DB'
+            if config.has_section(cfg_section):
+                hostname = config.get(cfg_section, 'hostname')
+                user = config.get(cfg_section, 'user')
+                auth = config.get(cfg_section, 'auth')
+                db_name = config.get(cfg_section, 'db_name')
+                if all([hostname, user, auth, db_name]):
+                    self.db = NewDBHandler()
+                    new_db_config = DBConfig(hostname, user, auth, db_name)
+                    self.conn = self.db.init_db_conn(new_db_config)
+
+            cfg_section = 'AMP'
+            if config.has_section(cfg_section):
+                self.qaqc_processor_source = config.get(cfg_section,
+                                                        'file_upload_source')
+
+            cfg_section = 'PHASE_I'
+            if config.has_section(cfg_section):
+                self.data_directory = config.get(cfg_section,
+                                                 'data_dir')
+
+        log_file_date = dt.datetime.now().strftime('%Y-%m-%d')
+        log_file_name = f'format_qaqc_driver_service_{log_file_date}.log'
+        log_dir = os.path.join(os.getcwd(), self.log_dir)
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
+        self.log_file_path = os.path.join(os.getcwd(),
+                                          self.log_dir,
+                                          log_file_name)
+        self.is_test = test
+        self.email_gen_path = './email_gen.py'
+        self.email_gen_team_path = './email_gen_team.py'
+        self.stale_count = 0
+
+    def recovery_process(self):
+        rerun_o_uuids = []
+        o_data_upload = \
+            self.db.get_undone_data_upload_log_o(
+                self.conn,
+                self.qaqc_processor_source,
+                self.lookback_h)
+        for row in o_data_upload:
+            uuid = row.get('upload_token')
+            if uuid not in rerun_o_uuids:
+                rerun_o_uuids.append(uuid)
+
+        ac_data_upload = \
+            self.db.get_undone_data_upload_log_ac(
+                self.conn,
+                self.qaqc_processor_source,
+                self.lookback_h)
+        rerun_uuids = []
+        for row in ac_data_upload:
+            comment = row.get('upload_comment')
+            if ('Archive upload for' in comment
+                    or 'repair candidate for' in comment):
+                process_id = comment.split()[-1]
+                while process_id:
+                    d = self.db.trace_o_data_upload(
+                        self.conn,
+                        process_id)
+                    if (not d.get('prior_process_id')
+                            and not d.get('zip_process_id')):
+                        process_id = d.get('log_id')
```

Comment:
For a zip file with children that have auto repaired files, I think this will result in the uuid of the originally uploaded zip file. If we just want to rerun the auto repaired files, I believe we just need the uuid of the uploaded archival contents?